tests/NCronJob.Tests/Helpers/EventsHelper.cs

```diff
@@ -0,0 +1,82 @@
+using Shouldly;
+
+namespace NCronJob.Tests;
+
+public static class EventsHelper
+{
+    public static IList<ExecutionProgress> WithOrchestrationId(
+        this IList<ExecutionProgress> events,
+        Guid orchestrationId)
+    {
+        return events.Where(e => e.CorrelationId == orchestrationId).ToList();
+    }
+
+    public static void AssertScheduledThenCancelled(IList<ExecutionProgress> events)
```

Comment:
Do we want to follow the shouldly style here?
AKA: `ScheduledThenCancelled(this ILIst<ExectuionProgress>)`?